rskj-core/src/main/java/co/rsk/signing/ECDSASignerFromDiskKey.java

```diff
@@ -0,0 +1,62 @@
+package co.rsk.signing;
+
+import org.ethereum.crypto.ECKey;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileNotFoundException;
+import java.util.List;
+
+/**
+ * Signer that reads key from disk.
+ *
+ * Key is identified by a key id.
+ *
+ * Has certain requirements on filesystem permissions
+ * of the file.
+ *
+ * No special authorization is required to sign a message.
+ *
+ * @author Ariel Mendelzon
+ */
+public class ECDSASignerFromDiskKey implements ECDSASigner {
+    private static final Logger logger = LoggerFactory.getLogger("ECDSASignerFromDiskKey");
+
+    private KeyId keyId;
+    private String keyPath;
+
+    public ECDSASignerFromDiskKey(KeyId keyId, String keyPath) {
+        this.keyId = keyId;
+        this.keyPath = keyPath;
+    }
+
+    @Override
+    public boolean canSignWith(KeyId keyId) {
+        return this.keyId.equals(keyId);
+    }
+
+    @Override
+    public ECDSASignerCheckResult check() {
+        KeyFileChecker checker = new KeyFileChecker(keyPath);
+
+        List<String> messages = checker.check();
+
+        return new ECDSASignerCheckResult(messages);
+    }
+
+    @Override
+    public ECDSASignature sign(KeyId keyId, Message message, SignAuthorization signAuthorization) throws FileNotFoundException {
+        if (!canSignWith(keyId)) {
+            logger.error("Can't sign with that key id. Requested {}", keyId);
+            throw new IllegalArgumentException(String.format("Can't sign with the requested signing key: %s", keyId));
+        }
+
+        // Read the key from disk
+        // We use Ethereum's ECKey for this and subsequent signing
+        // since it's good enough and there's no loss of generality
+        ECKey key = ECKey.fromPrivate(new KeyFileHandler(keyPath).privateKey());
```

Comment:
How about `return new KeyFileHandler(keyPath).sign(message.getBytes())`? This lets you hide the private key and usage of `ECKey` as an implementation detail of the `KeyFileHandler`.

Reply:
The `KeyFileHandler` is left as is from a previously-existing version (so is `KeyFileChecker`). As far as I'm concerned, its responsibility goes is limited to reading a key file in a known format from disk. Further specific key instance creation and signing is not in its scope (that's what this class does, actually).